File: Csharp.Minimal.Api/Routes/TeacherRoutes.cs
namespace Csharp.Minimal.Api.Routes;

/// <summary>
/// Defines the routes to the teacher endpoints.
/// </summary>
internal static class TeacherRoutes
{
    /// <summary>
    /// The route to the endpoint with the action:
    /// add teacher
    /// </summary>
    public const string Add = Base;
    
    /// <summary>
    /// The base route of all teacher endpoints.
    /// </summary>
    private const string Base = "/api/Teachers";
    
    /// <summary>
    /// The route to the endpoint with the action:
    /// delete teacher
    /// </summary>
    public const string Delete = Get;
    
    /// <summary>
    /// The route to the endpoint with the action:
    /// get teacher
    /// </summary>
    public const string Get = $"{Base}/{{id:guid}}";

    /// <summary>
    /// The route to the endpoint with the action:
    /// get all teachers
    /// </summary>
    public const string GetAll = Base;

    /// <summary>
    /// The route to the endpoint with the action:
    /// update teachers
    /// </summary>
    public const string Update = Get;
}

File: Fsharp.Minimal.Api/Routes/TeacherRoutes.fs
namespace Fsharp.Minimal.Api.Routes

[<RequireQualifiedAccess>]
module TeacherRoutes =

    /// <summary>
    /// The base route of all teacher endpoints.
    /// </summary>
    [<Literal>]
    let Base = "/api/Teachers"
    
    /// <summary>
    /// The route to the endpoint with the action:
    /// add teacher
    /// </summary>
    [<Literal>]
    let Add = Base
    
    /// <summary>
    /// The route to the endpoint with the action:
    /// get teacher
    /// </summary>
    [<Literal>]
    let Get = Base + "/{id:guid}"
    
    /// <summary>
    /// The route to the endpoint with the action:
    /// get all teachers
    /// </summary>
    [<Literal>]
    let GetAll = Base
    
    /// <summary>
    /// The route to the endpoint with the action:
    /// delete teacher
    /// </summary>
    [<Literal>]
    let Delete = Get
    
    /// <summary>
    /// The route to the endpoint with the action:
    /// update teacher
    /// </summary>
    [<Literal>]
    let Update = Get
